jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -284,6 +315,41 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
 
         return collapsed_spec
 
+    def cone_aperture(self):
+        # Retrieve flux cube and create an array to represent the cone mask
+        flux_cube = self._app._jdaviz_helper._loaded_flux_cube.get_object(cls=Spectrum1D,
+                                                                          statistic=None)
+        # TODO: Replace with code for retrieving display_unit in cubeviz when it is available
+        display_unit = flux_cube.spectral_axis.unit
+        if display_unit.physical_type != 'length':
+            self.hub.broadcast(SnackbarMessage('Spectral axis unit physical type is '
+                                               f'{display_unit.physical_type}, must be length',
+                                               color="error", sender=self))
+            return
+
+        masks_weights = np.zeros_like(flux_cube.flux.value, dtype=np.float32)
+        masks_bool_values = np.ones_like(flux_cube.flux.value, dtype=bool)
+
+        # Center is reverse coordinates
+        center = (self.aperture.selected_spatial_region.center.y,
+                  self.aperture.selected_spatial_region.center.x)
+
+        # Loop through cube and create cone aperture at each wavelength. Then convert that to a
+        # mask using the selected aperture method and add that to a mask cube.
+        # TODO: Use flux_cube.spectral_axis.to_value(display_unit) when we have unit conversion.
+        radii = ((flux_cube.spectral_axis.value / self.reference_wavelength) *
+                 self.aperture.selected_spatial_region.radius)
+        im_shape = (flux_cube.shape[0], flux_cube.shape[1])
+        aper_method = self.aperture_method_selected.lower()
+        for index, radius in enumerate(radii):
+            aperture = CircularAperture(center, r=radius)
+            slice_mask = aperture.to_mask(method=aper_method).to_image(im_shape)
+            # Send exact fractional pixel array in the `data` attribute
+            # and the boolean mask in the `mask` attribute
+            masks_bool_values[:, :, index] = ~(slice_mask > 0)
+            masks_weights[:, :, index] = slice_mask
+        return NDDataArray(data=masks_weights, mask=masks_bool_values)
```

Comment:
If the aperture is already encoded in the weights, so we still need a `mask=`? Let's wait for Brett to review.

Thanks!

Reply:
Is it more useful for this method to return just `slice_mask` or an `NDDataArray` object?

Reply:
I am not sure. From what Brett said, the benefit of encapsulating this as NDData is so the `*` operator would automatically propagate any uncertainty and/or mask, but photutils aperture does not give you any of those.